### PR TITLE
LEMS-494-refactor-events-into-divisions

### PR DESCRIPTION
**Pull request by johnmeshulam**
Co-authored-by: WiFi <itamaroryan@users.noreply.github.com>

### CODE REVIEW

File: libs/types/src/lib/websocket.ts [hunk 1]
```diff
@@ -199,37 +199,37 @@ export interface WSClientEmittedEvents {
...
+    divisionId: string,
     presentationId: string,
     newPresentationState: Partial<PresentationState>,
     callback: (response: { ok: boolean; error?: string }) => void
   ) => void;
 }
 
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface WSInterServerEvents {
+export interface WSInterServerDivisions {
```
Comment (itamaroryan) [inline libs/types/src/lib/websocket.ts:224]: Uhhhhh

File: libs/types/src/lib/websocket.ts [hunk 2]
```diff
@@ -199,37 +199,37 @@ export interface WSClientEmittedEvents {
...
   // ...
 }
 
 export interface WSSocketData {
   rooms: Array<WSRoomName>;
 }
 
-export interface WSEventListener {
-  name: EventNames<WSServerEmittedEvents> | EventNames<WSClientEmittedEvents>;
+export interface WSDivisionListener {
```
Comment (itamaroryan) [inline libs/types/src/lib/websocket.ts:232]: Hmmm

File: apps/backend/src/middlewares/websocket/division-validator.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+const divisionValidator = async (socket: Socket, next) => {
+  const divisionId = socket.nsp.name.split('/')[2];
+
+  if (!divisionId) {
+    return next(new Error('NO_EVENT_ID'));
+  }
+
+  const division = db.getDivision({ _id: new ObjectId(divisionId) });
+  if (!division) {
+    return next(new Error('EVENT_NOT_FOUND'));
```
Comment (johnmeshulam) [inline apps/backend/src/middlewares/websocket/division-validator.ts:14]: DIVISION

File: apps/frontend/pages/division/[divisionId]/tournament-manager.tsx [hunk 4]
```diff
@@ -141,20 +141,20 @@ const Page: NextPage<Props> = ({
...
-      { name: 'matchUpdated', handler: handleMatchEvent },
+      { name: 'judgingSessionStarted', handler: handleSessionDivision },
+      { name: 'judgingSessionCompleted', handler: handleSessionDivision },
+      { name: 'judgingSessionAborted', handler: handleSessionDivision },
+      { name: 'judgingSessionUpdated', handler: handleSessionDivision },
+      { name: 'matchLoaded', handler: handleMatchDivision },
+      { name: 'matchStarted', handler: handleMatchDivision },
+      { name: 'matchCompleted', handler: handleMatchDivision },
+      { name: 'matchAborted', handler: handleMatchDivision },
+      { name: 'matchUpdated', handler: handleMatchDivision },
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/tournament-manager.tsx:157]: Oh no

File: apps/frontend/pages/division/[divisionId]/tournament-manager.tsx [hunk 5]
```diff
@@ -112,9 +112,9 @@ const Page: NextPage<Props> = ({
     );
   };
 
-  const handleMatchEvent = (
+  const handleMatchDivision = (
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/tournament-manager.tsx:115]: Fix this

File: apps/frontend/pages/division/[divisionId]/tournament-manager.tsx [hunk 6]
```diff
@@ -101,7 +101,7 @@ const Page: NextPage<Props> = ({
     );
   };
 
-  const handleSessionEvent = (session: WithId<JudgingSession>) => {
+  const handleSessionDivision = (session: WithId<JudgingSession>) => {
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/tournament-manager.tsx:104]: Fix this

File: apps/frontend/pages/division/[divisionId]/scorekeeper.tsx [hunk 7]
```diff
@@ -59,24 +59,27 @@ const Page: NextPage<Props> = ({
...
         });
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [nextMatch]);
 
-  const handleMatchEvent = (match: WithId<RobotGameMatch>, newEventState?: WithId<EventState>) => {
+  const handleMatchDivision = (
+    match: WithId<RobotGameMatch>,
+    newDivisionState?: WithId<DivisionState>
+  ) => {
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/scorekeeper.tsx:82]: Rename this

File: apps/frontend/pages/division/[divisionId]/scorekeeper.tsx [hunk 8]
```diff
@@ -115,17 +118,17 @@ const Page: NextPage<Props> = ({
...
+      { name: 'matchStarted', handler: handleMatchDivision },
       { name: 'matchAborted', handler: handleMatchAborted },
-      { name: 'matchCompleted', handler: handleMatchEvent },
-      { name: 'matchUpdated', handler: handleMatchEvent },
-      { name: 'audienceDisplayUpdated', handler: setEventState },
-      { name: 'presentationUpdated', handler: setEventState },
+      { name: 'matchCompleted', handler: handleMatchDivision },
+      { name: 'matchUpdated', handler: handleMatchDivision },
+      { name: 'audienceDisplayUpdated', handler: setDivisionState },
+      { name: 'presentationUpdated', handler: setDivisionState },
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/scorekeeper.tsx:131]: Refactor accordingly

File: apps/frontend/pages/division/[divisionId]/reports/scoreboard.tsx [hunk 9]
```diff
@@ -18,25 +18,25 @@ import { compareScoreArrays } from '@lems/utils/arrays';
...
   scoresheets: initialScoresheets
 }) => {
   const router = useRouter();
   const [scoresheets, setScoresheets] = useState<Array<WithId<Scoresheet>>>(initialScoresheets);
 
   const teamNumberComparator: GridComparatorFn<Team> = (v1, v2) => v1.number - v2.number;
 
-  const handleScoresheetEvent = (scoresheet: WithId<Scoresheet>) => {
+  const handleScoresheetDivision = (scoresheet: WithId<Scoresheet>) => {
     setScoresheets(scoresheets =>
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/scoreboard.tsx:40]: Rename this

File: apps/frontend/pages/division/[divisionId]/reports/scoreboard.tsx [hunk 10]
```diff
@@ -47,22 +47,22 @@ const Page: NextPage<Props> = ({
     );
   };
 
-  const { connectionStatus } = useWebsocket(event._id.toString(), ['field'], undefined, [
-    { name: 'scoresheetUpdated', handler: handleScoresheetEvent }
+  const { connectionStatus } = useWebsocket(division._id.toString(), ['field'], undefined, [
+    { name: 'scoresheetUpdated', handler: handleScoresheetDivision }
   ]);
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/scoreboard.tsx:52]: Refactor accordingly

File: apps/backend/src/websocket/index.ts [hunk 11]
```diff
@@ -2,7 +2,7 @@ import { Socket } from 'socket.io';
 import {
   WSServerEmittedEvents,
   WSClientEmittedEvents,
-  WSInterServerEvents,
+  WSInterServerDivisions,
```
Comment (johnmeshulam) [inline apps/backend/src/websocket/index.ts:5]: WSInterServerEvents

File: apps/backend/src/websocket/index.ts [hunk 12]
```diff
@@ -29,12 +29,12 @@ import {
 import { handleUpdateAudienceDisplay, handleUpdatePresentation } from './handlers/audience-display';
 
 const websocket = (
-  socket: Socket<WSClientEmittedEvents, WSServerEmittedEvents, WSInterServerEvents, WSSocketData>
```
Comment (johnmeshulam) [inline apps/backend/src/websocket/index.ts:32]: WSInterServerEvents

File: apps/frontend/pages/division/[divisionId]/reports/judging-status.tsx [hunk 13]
```diff
@@ -200,9 +200,9 @@ const Page: NextPage<Props> = ({
     );
   };
 
-  const handleSessionEvent = (
+  const handleSessionDivision = (
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/judging-status.tsx:203]: Rename this

File: apps/frontend/pages/division/[divisionId]/reports/judging-status.tsx [hunk 14]
```diff
@@ -213,10 +213,13 @@ const Page: NextPage<Props> = ({
...
 
-    if (newEventState) setEventState(newEventState);
+    if (newDivisionState) setDivisionState(newDivisionState);
   };
 
-  const handleMatchEvent = (match: WithId<RobotGameMatch>, newEventState?: WithId<EventState>) => {
+  const handleMatchDivision = (
+    match: WithId<RobotGameMatch>,
+    newDivisionState?: WithId<DivisionState>
+  ) => {
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/judging-status.tsx:222]: Rename this

File: apps/frontend/pages/division/[divisionId]/reports/judging-status.tsx [hunk 15]
```diff
@@ -226,54 +229,54 @@ const Page: NextPage<Props> = ({
...
-      { name: 'matchLoaded', handler: handleMatchEvent },
-      { name: 'matchStarted', handler: handleMatchEvent },
-      { name: 'matchAborted', handler: handleMatchEvent },
-      { name: 'matchCompleted', handler: handleMatchEvent },
-      { name: 'matchUpdated', handler: handleMatchEvent }
+      { name: 'matchLoaded', handler: handleMatchDivision },
+      { name: 'matchStarted', handler: handleMatchDivision },
+      { name: 'matchAborted', handler: handleMatchDivision },
+      { name: 'matchCompleted', handler: handleMatchDivision },
+      { name: 'matchUpdated', handler: handleMatchDivision }
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/judging-status.tsx:249]: Refactor accordingly

File: apps/frontend/pages/division/[divisionId]/reports/judging-schedule.tsx [hunk 16]
```diff
@@ -45,7 +45,7 @@ const Page: NextPage<Props> = ({
     );
   };
 
-  const handleSessionEvent = (session: WithId<JudgingSession>) => {
+  const handleSessionDivision = (session: WithId<JudgingSession>) => {
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/judging-schedule.tsx:48]: Rename this

File: apps/frontend/pages/division/[divisionId]/reports/judging-schedule.tsx [hunk 17]
```diff
@@ -57,12 +57,12 @@ const Page: NextPage<Props> = ({
...
 
   const { connectionStatus } = useWebsocket(
-    event._id.toString(),
+    division._id.toString(),
     ['pit-admin', 'judging'],
     undefined,
     [
       { name: 'teamRegistered', handler: handleTeamRegistered },
-      { name: 'judgingSessionUpdated', handler: handleSessionEvent }
+      { name: 'judgingSessionUpdated', handler: handleSessionDivision }
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/judging-schedule.tsx:65]: Refactor accordingly

File: apps/frontend/pages/division/[divisionId]/reports/field-timer.tsx [hunk 18]
```diff
@@ -58,7 +65,10 @@ const Page: NextPage<Props> = ({
   const getCountdownTarget = (startTime: Date) =>
     dayjs(startTime).add(MATCH_LENGTH, 'seconds').toDate();
 
-  const handleMatchEvent = (match: WithId<RobotGameMatch>, newEventState?: WithId<EventState>) => {
+  const handleMatchDivision = (
+    match: WithId<RobotGameMatch>,
+    newDivisionState?: WithId<DivisionState>
+  ) => {
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/field-timer.tsx:71]: Rename this

File: apps/frontend/pages/division/[divisionId]/reports/field-timer.tsx [hunk 19]
```diff
@@ -68,35 +78,35 @@ const Page: NextPage<Props> = ({
...
-  const { connectionStatus } = useWebsocket(event._id.toString(), ['field'], undefined, [
+  const { connectionStatus } = useWebsocket(division._id.toString(), ['field'], undefined, [
     {
       name: 'matchStarted',
-      handler: (newMatch, newEventState) => {
-        if (eventState.audienceDisplay.screen === 'scores') sounds.current.start.play();
-        handleMatchEvent(newMatch, newEventState);
+      handler: (newMatch, newDivisionState) => {
+        if (divisionState.audienceDisplay.screen === 'scores') sounds.current.start.play();
+        handleMatchDivision(newMatch, newDivisionState);
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/field-timer.tsx:89]: Refactor this

File: apps/frontend/pages/division/[divisionId]/reports/field-timer.tsx [hunk 20]
```diff
@@ -68,35 +78,35 @@ const Page: NextPage<Props> = ({
...
       }
     },
     {
       name: 'matchAborted',
-      handler: (newMatch, newEventState) => {
-        if (eventState.audienceDisplay.screen === 'scores') sounds.current.abort.play();
-        handleMatchEvent(newMatch, newEventState);
+      handler: (newMatch, newDivisionState) => {
+        if (divisionState.audienceDisplay.screen === 'scores') sounds.current.abort.play();
+        handleMatchDivision(newMatch, newDivisionState);
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/field-timer.tsx:96]: Refactor this

File: apps/frontend/pages/division/[divisionId]/reports/field-timer.tsx [hunk 21]
```diff
@@ -68,35 +78,35 @@ const Page: NextPage<Props> = ({
...
       }
     },
     {
       name: 'matchCompleted',
-      handler: (newMatch, newEventState) => {
-        if (eventState.audienceDisplay.screen === 'scores') sounds.current.end.play();
-        handleMatchEvent(newMatch, newEventState);
+      handler: (newMatch, newDivisionState) => {
+        if (divisionState.audienceDisplay.screen === 'scores') sounds.current.end.play();
+        handleMatchDivision(newMatch, newDivisionState);
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/field-timer.tsx:109]: Refactor this

File: apps/frontend/pages/division/[divisionId]/reports/field-status.tsx [hunk 22]
```diff
@@ -150,7 +150,10 @@ const Page: NextPage<Props> = ({
     );
   };
 
-  const handleMatchEvent = (match: WithId<RobotGameMatch>, newEventState?: WithId<EventState>) => {
+  const handleMatchDivision = (
+    match: WithId<RobotGameMatch>,
+    newDivisionState?: WithId<DivisionState>
+  ) => {
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/field-status.tsx:156]: Rename this

File: apps/frontend/pages/division/[divisionId]/reports/field-status.tsx [hunk 23]
```diff
@@ -160,12 +163,12 @@ const Page: NextPage<Props> = ({
       })
     );
 
-    if (newEventState) setEventState(newEventState);
+    if (newDivisionState) setDivisionState(newDivisionState);
   };
 
-  const handleSessionEvent = (
+  const handleSessionDivision = (
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/field-status.tsx:169]: Rename this

File: apps/frontend/pages/division/[divisionId]/reports/field-status.tsx [hunk 24]
```diff
@@ -176,24 +179,24 @@ const Page: NextPage<Props> = ({
...
-      { name: 'judgingSessionUpdated', handler: handleSessionEvent }
+      { name: 'matchLoaded', handler: handleMatchDivision },
+      { name: 'matchStarted', handler: handleMatchDivision },
+      { name: 'matchAborted', handler: handleMatchDivision },
+      { name: 'matchCompleted', handler: handleMatchDivision },
+      { name: 'matchUpdated', handler: handleMatchDivision },
+      { name: 'judgingSessionStarted', handler: handleSessionDivision },
+      { name: 'judgingSessionCompleted', handler: handleSessionDivision },
+      { name: 'judgingSessionAborted', handler: handleSessionDivision },
+      { name: 'judgingSessionUpdated', handler: handleSessionDivision }
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/field-status.tsx:199]: Refactor accordingly

File: apps/frontend/pages/division/[divisionId]/reports/field-schedule.tsx [hunk 25]
```diff
@@ -48,7 +48,7 @@ const Page: NextPage<Props> = ({
     );
   };
 
-  const handleMatchEvent = (match: WithId<RobotGameMatch>) => {
+  const handleMatchDivision = (match: WithId<RobotGameMatch>) => {
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/field-schedule.tsx:51]: Rename

File: apps/frontend/pages/division/[divisionId]/reports/field-schedule.tsx [hunk 26]
```diff
@@ -59,9 +59,9 @@ const Page: NextPage<Props> = ({
     );
   };
 
-  const { connectionStatus } = useWebsocket(event._id.toString(), ['pit-admin'], undefined, [
+  const { connectionStatus } = useWebsocket(division._id.toString(), ['pit-admin'], undefined, [
     { name: 'teamRegistered', handler: handleTeamRegistered },
-    { name: 'matchUpdated', handler: handleMatchEvent }
+    { name: 'matchUpdated', handler: handleMatchDivision }
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/reports/field-schedule.tsx:64]: Refactor

File: apps/frontend/pages/division/[divisionId]/referee/index.tsx [hunk 27]
```diff
@@ -50,12 +50,12 @@ const Page: NextPage<Props> = ({
     );
   };
 
-  const handleMatchEvent = (
+  const handleMatchDivision = (
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/referee/index.tsx:53]: Rename

File: apps/frontend/pages/division/[divisionId]/referee/index.tsx [hunk 28]
```diff
@@ -75,14 +75,14 @@ const Page: NextPage<Props> = ({
...
     undefined,
     [
-      { name: 'matchLoaded', handler: handleMatchEvent },
-      { name: 'matchStarted', handler: handleMatchEvent },
-      { name: 'matchAborted', handler: handleMatchEvent },
-      { name: 'matchCompleted', handler: handleMatchEvent },
+      { name: 'matchLoaded', handler: handleMatchDivision },
+      { name: 'matchStarted', handler: handleMatchDivision },
+      { name: 'matchAborted', handler: handleMatchDivision },
+      { name: 'matchCompleted', handler: handleMatchDivision },
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/referee/index.tsx:85]: Refactor

File: apps/frontend/pages/division/[divisionId]/queuer.tsx [hunk 29]
```diff
@@ -56,11 +56,14 @@ const Page: NextPage<Props> = ({
...
-  const [eventState, setEventState] = useState<WithId<EventState>>(initialEventState);
+  const [divisionState, setDivisionState] = useState<WithId<DivisionState>>(initialDivisionState);
   const [matches, setMatches] = useState<Array<WithId<RobotGameMatch>>>(initialMatches);
   const [sessions, setSessions] = useState<Array<WithId<JudgingSession>>>(initialSessions);
 
-  const handleMatchEvent = (match: WithId<RobotGameMatch>, newEventState?: WithId<EventState>) => {
+  const handleMatchDivision = (
+    match: WithId<RobotGameMatch>,
+    newDivisionState?: WithId<DivisionState>
+  ) => {
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/queuer.tsx:66]: Rename

File: apps/frontend/pages/division/[divisionId]/queuer.tsx [hunk 30]
```diff
@@ -84,9 +87,9 @@ const Page: NextPage<Props> = ({
     );
   };
 
-  const handleSessionEvent = (
+  const handleSessionDivision = (
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/queuer.tsx:90]: Rename

File: apps/frontend/pages/division/[divisionId]/queuer.tsx [hunk 31]
```diff
@@ -97,60 +100,64 @@ const Page: NextPage<Props> = ({
...
+    { name: 'matchUpdated', handler: handleMatchDivision },
     { name: 'teamRegistered', handler: handleTeamRegistered },
-    { name: 'judgingSessionStarted', handler: handleSessionEvent },
-    { name: 'judgingSessionCompleted', handler: handleSessionEvent },
-    { name: 'judgingSessionAborted', handler: handleSessionEvent },
-    { name: 'judgingSessionUpdated', handler: handleSessionEvent }
+    { name: 'judgingSessionStarted', handler: handleSessionDivision },
+    { name: 'judgingSessionCompleted', handler: handleSessionDivision },
+    { name: 'judgingSessionAborted', handler: handleSessionDivision },
+    { name: 'judgingSessionUpdated', handler: handleSessionDivision }
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/queuer.tsx:115]: Refactor

File: apps/frontend/hooks/use-websocket.ts [hunk 32]
```diff
@@ -1,14 +1,14 @@
 import { useState, useEffect } from 'react';
-import { WSEventListener, ConnectionStatus, WSRoomName } from '@lems/types';
+import { WSDivisionListener, ConnectionStatus, WSRoomName } from '@lems/types';
```
Comment (johnmeshulam) [inline apps/frontend/hooks/use-websocket.ts:2]: WSEventListener

File: apps/frontend/hooks/use-websocket.ts [hunk 33]
```diff
@@ -1,14 +1,14 @@
...
 import { getSocket } from '../lib/utils/websocket';
 
 export const useWebsocket = (
-  eventId: string,
+  divisionId: string,
   rooms: Array<WSRoomName>,
   init?: (...args: any[]) => void | Promise<void>,
-  wsevents?: Array<WSEventListener>
+  wsevents?: Array<WSDivisionListener>
```
Comment (johnmeshulam) [inline apps/frontend/hooks/use-websocket.ts:9]: WSEventListener

File: apps/frontend/pages/division/[divisionId]/mc.tsx [hunk 34]
```diff
@@ -50,7 +57,10 @@ const Page: NextPage<Props> = ({
     );
   };
 
-  const handleMatchEvent = (match: WithId<RobotGameMatch>, newEventState?: WithId<EventState>) => {
+  const handleMatchDivision = (
+    match: WithId<RobotGameMatch>,
+    newDivisionState?: WithId<DivisionState>
+  ) => {
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/mc.tsx:63]: Rename

File: apps/frontend/pages/division/[divisionId]/mc.tsx [hunk 35]
```diff
@@ -60,34 +70,34 @@ const Page: NextPage<Props> = ({
...
-    { name: 'matchUpdated', handler: handleMatchEvent },
-    { name: 'audienceDisplayUpdated', handler: setEventState },
-    { name: 'presentationUpdated', handler: setEventState }
+    { name: 'matchLoaded', handler: handleMatchDivision },
+    { name: 'matchStarted', handler: handleMatchDivision },
+    { name: 'matchAborted', handler: handleMatchDivision },
+    { name: 'matchCompleted', handler: handleMatchDivision },
+    { name: 'matchUpdated', handler: handleMatchDivision },
+    { name: 'audienceDisplayUpdated', handler: setDivisionState },
+    { name: 'presentationUpdated', handler: setDivisionState }
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/mc.tsx:84]: Refacotr

File: apps/frontend/pages/division/[divisionId]/lead-judge.tsx [hunk 36]
```diff
@@ -51,7 +51,10 @@ const Page: NextPage<Props> = ({
   const [sessions, setSessions] = useState<Array<WithId<JudgingSession>>>(initialSessions);
   const [rubrics, setRubrics] = useState<Array<WithId<Rubric<JudgingCategory>>>>(initialRubrics);
 
-  const handleSessionEvent = (session: WithId<JudgingSession>, eventState?: WithId<EventState>) => {
+  const handleSessionDivision = (
+    session: WithId<JudgingSession>,
+    divisionState?: WithId<DivisionState>
+  ) => {
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/lead-judge.tsx:57]: Rename

File: apps/frontend/pages/division/[divisionId]/lead-judge.tsx [hunk 37]
```diff
@@ -93,14 +96,14 @@ const Page: NextPage<Props> = ({
...
     undefined,
     [
-      { name: 'judgingSessionStarted', handler: handleSessionEvent },
-      { name: 'judgingSessionCompleted', handler: handleSessionEvent },
-      { name: 'judgingSessionAborted', handler: handleSessionEvent },
-      { name: 'judgingSessionUpdated', handler: handleSessionEvent },
+      { name: 'judgingSessionStarted', handler: handleSessionDivision },
+      { name: 'judgingSessionCompleted', handler: handleSessionDivision },
+      { name: 'judgingSessionAborted', handler: handleSessionDivision },
+      { name: 'judgingSessionUpdated', handler: handleSessionDivision },
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/lead-judge.tsx:106]: Refacotr

File: apps/frontend/pages/division/[divisionId]/audience-display.tsx [hunk 38]
```diff
@@ -86,15 +86,15 @@ const Page: NextPage<Props> = ({
     );
   };
 
-  const handleMatchEvent = (
+  const handleMatchDivision = (
```
Comment (johnmeshulam) [inline apps/frontend/pages/division/[divisionId]/audience-display.tsx:89]: handleMatchEvent

File: apps/frontend/pages/division/[divisionId]/audience-display.tsx [hunk 39]
```diff
@@ -86,15 +86,15 @@ const Page: NextPage<Props> = ({
...
     newMatch: WithId<RobotGameMatch>,
-    newEventState?: WithId<EventState>
+    newDivisionState?: WithId<DivisionState>
   ) => {
     updateMatches(newMatch);
-    if (newEventState) setEventState(newEventState);
+    if (newDivisionState) setDivisionState(newDivisionState);
   };
 
-  const handleScoresheetEvent = (scoresheet: WithId<Scoresheet>) => {
```
Comment (johnmeshulam) [inline apps/frontend/pages/division/[divisionId]/audience-display.tsx:97]: handleScoresheetEvent

File: apps/frontend/pages/division/[divisionId]/judge.tsx [hunk 40]
```diff
@@ -61,7 +61,10 @@ const Page: NextPage<Props> = ({
       : ({} as WithId<Team>);
   }, [teams, currentSession]);
 
-  const handleSessionEvent = (session: WithId<JudgingSession>, eventState?: WithId<EventState>) => {
+  const handleSessionDivision = (
+    session: WithId<JudgingSession>,
+    divisionState?: WithId<DivisionState>
+  ) => {
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/judge.tsx:67]: Rename this

File: apps/frontend/pages/division/[divisionId]/audience-display.tsx [hunk 41]
```diff
@@ -122,39 +122,39 @@ const Page: NextPage<Props> = ({
...
-  useWebsocket(event._id.toString(), ['pit-admin', 'field', 'audience-display'], undefined, [
+  useWebsocket(division._id.toString(), ['pit-admin', 'field', 'audience-display'], undefined, [
     {
       name: 'matchStarted',
-      handler: (newMatch, newEventState) => {
-        if (eventState.audienceDisplay.screen === 'scores') sounds.current.start.play();
-        handleMatchEvent(newMatch, newEventState);
+      handler: (newMatch, newDivisionState) => {
+        if (divisionState.audienceDisplay.screen === 'scores') sounds.current.start.play();
+        handleMatchDivision(newMatch, newDivisionState);
```
Comment (johnmeshulam) [inline apps/frontend/pages/division/[divisionId]/audience-display.tsx:130]: handleMatchEvent

File: apps/frontend/pages/division/[divisionId]/audience-display.tsx [hunk 42]
```diff
@@ -122,39 +122,39 @@ const Page: NextPage<Props> = ({
...
       }
     },
     {
       name: 'matchAborted',
-      handler: (newMatch, newEventState) => {
-        if (eventState.audienceDisplay.screen === 'scores') sounds.current.abort.play();
-        handleMatchEvent(newMatch, newEventState);
+      handler: (newMatch, newDivisionState) => {
+        if (divisionState.audienceDisplay.screen === 'scores') sounds.current.abort.play();
+        handleMatchDivision(newMatch, newDivisionState);
```
Comment (johnmeshulam) [inline apps/frontend/pages/division/[divisionId]/audience-display.tsx:137]: handleMatchEvent

File: apps/frontend/pages/division/[divisionId]/judge.tsx [hunk 43]
```diff
@@ -95,14 +98,14 @@ const Page: NextPage<Props> = ({
...
     undefined,
     [
-      { name: 'judgingSessionStarted', handler: handleSessionEvent },
-      { name: 'judgingSessionCompleted', handler: handleSessionEvent },
-      { name: 'judgingSessionAborted', handler: handleSessionEvent },
-      { name: 'judgingSessionUpdated', handler: handleSessionEvent },
+      { name: 'judgingSessionStarted', handler: handleSessionDivision },
+      { name: 'judgingSessionCompleted', handler: handleSessionDivision },
+      { name: 'judgingSessionAborted', handler: handleSessionDivision },
+      { name: 'judgingSessionUpdated', handler: handleSessionDivision },
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/judge.tsx:108]: Refactor that

File: apps/frontend/pages/division/[divisionId]/audience-display.tsx [hunk 44]
```diff
@@ -122,39 +122,39 @@ const Page: NextPage<Props> = ({
...
       }
     },
     {
       name: 'matchCompleted',
-      handler: (newMatch, newEventState) => {
-        if (eventState.audienceDisplay.screen === 'scores') sounds.current.end.play();
-        handleMatchEvent(newMatch, newEventState);
+      handler: (newMatch, newDivisionState) => {
+        if (divisionState.audienceDisplay.screen === 'scores') sounds.current.end.play();
+        handleMatchDivision(newMatch, newDivisionState);
```
Comment (johnmeshulam) [inline apps/frontend/pages/division/[divisionId]/audience-display.tsx:150]: handleMatchEvent for EVNERYTHING opn this page

Comment (johnmeshulam) [inline apps/frontend/pages/division/[divisionId]/audience-display.tsx:150]: handleMatchEvent for EVNERYTHING opn this page

File: apps/frontend/pages/division/[divisionId]/judge-advisor.tsx [hunk 45]
```diff
@@ -75,7 +75,7 @@ const Page: NextPage<Props> = ({
     return a.place - b.place;
   });
 
-  const handleSessionEvent = (session: WithId<JudgingSession>) => {
+  const handleSessionDivision = (session: WithId<JudgingSession>) => {
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/judge-advisor.tsx:78]: Rename

File: apps/frontend/pages/division/[divisionId]/judge-advisor.tsx [hunk 46]
```diff
@@ -123,14 +123,14 @@ const Page: NextPage<Props> = ({
...
     undefined,
     [
-      { name: 'judgingSessionStarted', handler: handleSessionEvent },
-      { name: 'judgingSessionCompleted', handler: handleSessionEvent },
-      { name: 'judgingSessionAborted', handler: handleSessionEvent },
-      { name: 'judgingSessionUpdated', handler: handleSessionEvent },
+      { name: 'judgingSessionStarted', handler: handleSessionDivision },
+      { name: 'judgingSessionCompleted', handler: handleSessionDivision },
+      { name: 'judgingSessionAborted', handler: handleSessionDivision },
+      { name: 'judgingSessionUpdated', handler: handleSessionDivision },
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/judge-advisor.tsx:133]: Refacotr this

File: apps/frontend/pages/division/[divisionId]/head-queuer.tsx [hunk 47]
```diff
@@ -49,28 +49,31 @@ const Page: NextPage<Props> = ({
...
+    [sessions, divisionState]
   );
   const nextSessions = useMemo(
-    () => sessions.filter(session => session.number === eventState.currentSession + 1),
-    [sessions, eventState]
+    () => sessions.filter(session => session.number === divisionState.currentSession + 1),
+    [sessions, divisionState]
   );
 
-  const handleMatchEvent = (match: WithId<RobotGameMatch>, newEventState?: WithId<EventState>) => {
```
Comment (johnmeshulam) [inline apps/frontend/pages/division/[divisionId]/head-queuer.tsx:73]: handleMatchEvent

File: apps/frontend/pages/division/[divisionId]/head-queuer.tsx [hunk 48]
```diff
@@ -94,9 +97,9 @@ const Page: NextPage<Props> = ({
     );
   };
 
-  const handleSessionEvent = (
```
Comment (johnmeshulam) [inline apps/frontend/pages/division/[divisionId]/head-queuer.tsx:97]: handleSessionEvent

File: apps/frontend/pages/division/[divisionId]/head-queuer.tsx [hunk 49]
```diff
@@ -107,23 +110,23 @@ const Page: NextPage<Props> = ({
...
+    if (newDivisionState) setDivisionState(newDivisionState);
   };
 
   const { socket, connectionStatus } = useWebsocket(
-    event._id.toString(),
+    division._id.toString(),
     ['field', 'pit-admin', 'judging'],
     undefined,
     [
-      { name: 'matchLoaded', handler: handleMatchEvent },
```
Comment (johnmeshulam) [inline apps/frontend/pages/division/[divisionId]/head-queuer.tsx:118]: correct for this entire WS block

File: apps/frontend/pages/division/[divisionId]/head-referee.tsx [hunk 50]
```diff
@@ -77,12 +77,12 @@ const Page: NextPage<Props> = ({
     );
   };
 
-  const handleMatchEvent = (
+  const handleMatchDivision = (
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/head-referee.tsx:80]: Rename

File: apps/frontend/pages/division/[divisionId]/head-referee.tsx [hunk 51]
```diff
@@ -123,15 +123,15 @@ const Page: NextPage<Props> = ({
...
-      { name: 'matchLoaded', handler: handleMatchEvent },
-      { name: 'matchStarted', handler: handleMatchEvent },
-      { name: 'matchCompleted', handler: handleMatchEvent },
-      { name: 'matchAborted', handler: handleMatchEvent },
-      { name: 'matchUpdated', handler: handleMatchEvent },
+      { name: 'matchLoaded', handler: handleMatchDivision },
+      { name: 'matchStarted', handler: handleMatchDivision },
+      { name: 'matchCompleted', handler: handleMatchDivision },
+      { name: 'matchAborted', handler: handleMatchDivision },
+      { name: 'matchUpdated', handler: handleMatchDivision },
```
Comment (itamaroryan) [inline apps/frontend/pages/division/[divisionId]/head-referee.tsx:134]: Refactor

**Pull request merged**